java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/ShareFileService.java

```diff
@@ -0,0 +1,243 @@
+		ShareFileRepository.ShareFile existingFile = getFileByOwnerAndFilepath(userFile.userid, userFile.name);
+		if (existingFile != null) {
+			existingFile.setUuid(uniqueCode);
+			shareFileRepository.saveAndFlush(existingFile);
+			return uniqueCode;
+		}
+		createShareFile(userFile, publicAccess, uniqueCode);
+
+		return uniqueCode;
+	}
+
+	private String generateUniqueCode() {
+		String uniqueCode = UUID.randomUUID().toString();
+		ShareFileRepository.ShareFile file = shareFileRepository.findByUuid(uniqueCode);
+		while (file != null) {
+			uniqueCode = UUID.randomUUID().toString();
+			file = shareFileRepository.findByUuid(uniqueCode);
+		}
+		return uniqueCode;
+	}
+
+	@Transactional
+	public ShareFileRepository.ShareFile createShareFile(PremiumUserFilesRepository.UserFile userFile, boolean publicAccess, String uniqueCode) {
+		ShareFileRepository.ShareFile shareFile = new ShareFileRepository.ShareFile();
+
+		String name = (userFile.name.lastIndexOf("/") >= 0)
+				? userFile.name.substring(userFile.name.lastIndexOf("/") + 1)
+				: userFile.name;
+
+		shareFile.setUuid(uniqueCode);
+		shareFile.setName(name);
+		shareFile.setFilepath(userFile.name);
+		shareFile.setType(userFile.type);
+		shareFile.setOwnerid(userFile.userid);
+		shareFile.setPublicAccess(publicAccess);
+
+		shareFileRepository.saveAndFlush(shareFile);
+		return shareFile;
+	}
+
+	public ShareFileRepository.ShareFile getFileByOwnerAndFilepath(int ownerid, String filepath) {
+		return shareFileRepository.findByOwneridAndFilepath(ownerid, filepath);
+	}
+
+	public List<ShareFileRepository.ShareFile> getFilesByOwner(int ownerid) {
+		return shareFileRepository.findByOwnerid(ownerid);
+	}
+
+	public ShareFileRepository.ShareFile getFileById(long id) {
+		return shareFileRepository.findById(id).orElse(null);
+	}
+
+	@Transactional
+	public boolean editAccessList(ShareFileRepository.ShareFile shareFile, Map<Integer, String> accessMap) {
+		List<ShareFileRepository.ShareFilesAccess> accessList = shareFile.getAccessRecords();
+		if (accessList == null) {
+			accessList = new ArrayList<>();
+		}
+		accessList.forEach(access -> {
+			String accessType = accessMap.get(access.getUser().id);
+			if (accessType != null) {
+				access.setAccess(accessType);
+			}
+		});
+		shareFile.setAccessRecords(accessList);
+		shareFileRepository.saveAndFlush(shareFile);
+		return true;
+	}
+
+	@Transactional
+	public boolean changeFileShareType(ShareFileRepository.ShareFile shareFile, String shareType) {
+		if (shareType.equals(PRIVATE_SHARE_TYPE)) {
+			shareFileRepository.delete(shareFile);
+		} else {
+			shareFile.setPublicAccess(shareType.equals(PUBLIC_SHARE_TYPE));
+			if (shareFile.isPublicAccess() && shareFile.getUuid() == null) {
+				String uuid = generateUniqueCode();
+				shareFile.setUuid(uuid);
+			}
+			shareFileRepository.saveAndFlush(shareFile);
+		}
+		return true;
+	}
+
+	public ResponseEntity<String> checkAccess(ShareFileRepository.ShareFile file) {
```

Comment:
`null` is unclear success return result for method named `checkAccess`

it sounds as `null = failed = no access`

please consider another approach